numcodecs/bitround.py

```diff
@@ -63,8 +63,8 @@ def encode(self, buf):
         maskbits = bits - self.keepbits
```

Comment:
```suggestion
        b = b.copy()
        maskbits = bits - self.keepbits
```

Reply:
[This](https://github.com/zarr-developers/numcodecs/pull/299#discussion_r845319562) seems clean

Reply:
nevermind, this was giving weird behavior. Explicit copy it is.